orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,207 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
 	// UUID was obtained from calling WMI infrastructure
 	return uuid, UUIDSourceWMI, nil
 }
+
+// getExecutablePath returns the current working directory
+func getExecutablePath() (string, error) {
+	// getting current executable fullpath
+	exec, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	// returns the current executable directory
+	return filepath.Dir(exec), nil
+}
+
+// getOrbitVersion returns the version of the Orbit executable
+func getOrbitVersion(path string) (string, error) {
+	const (
+		expectedPrefix      = "orbit "
+		expectedVersionFlag = "-version"
+	)
+
+	if len(path) == 0 {
+		return "", errors.New("input executable is empty")
+	}
+
+	// running the executable with the version flag
+	args := []string{expectedVersionFlag}
+	out, err := exec.Command(path, args...).Output()
+	if err != nil {
+		return "", fmt.Errorf("there was a problem running target executable: %w", err)
+	}
+
+	// parsing the output
+	versionOutputStr := string(out)
+	if len(versionOutputStr) == 0 {
+		return "", errors.New("empty executable output")
+	}
+
+	outputByLines := strings.Split(strings.TrimRight(versionOutputStr, "\n"), "\n")
+	if len(outputByLines) < 1 {
+		return "", errors.New("expected number of lines is not present")
+	}
+
+	rawVersionStr := strings.TrimSpace(strings.ToLower(outputByLines[0]))
+	if !strings.HasPrefix(rawVersionStr, expectedPrefix) {
+		return "", errors.New("expected version prefix is not present")
+	}
+
+	// getting the actual version string
+	versionStr := strings.TrimPrefix(rawVersionStr, expectedPrefix)
+	if len(versionStr) == 0 {
+		return "", errors.New("expected version information is not present")
+	}
+
+	return versionStr, nil
+}
+
+// versionCheckForfixSymlinkNotPresentQuirk checks if the target orbit version has the problematic logic
+func versionCheckForfixSymlinkNotPresentQuirk(orbitPath string) (bool, error) {
```

Comment:
This always returns either `false, err` or `true, nil` (so I believe you can drop the boolean and return an error)

Reply:
good, just fixed that